presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -169,6 +169,35 @@ public static int getFieldIndex(MessageType fileSchema, String name)
         return type;
     }
 
+    // Find the column index by name following the same logic as findParquetTypeByName
+    public static int findFieldIndexByName(MessageType fileSchema, String name)
```

Comment:
I would suggest doing the same way as the `findParquetTypeByName` to reduce the code duplication.
```
{
  int fieldIndex = getFieldIndex(fileSchema, name);
  
  // when a parquet field is a hive keyword we append an _ to it in hive. When doing
  // a name-based lookup, we need to strip it off again if we didn't get a direct match.
  if (fieldIndex == -1 && name.endsWith("_")) {
    fieldIndex = getFieldIndex(fileSchema, name.substring(0, name.length() - 1));
  }

  return fieldIndex;
}
```